### DISCUSSION TITLE
Hosting in Termux on Android

### DISCUSSION THREAD

**Discussion by bobloy** [Cog-Creators/Red-DiscordBot#4582] [General]
At the moment Termux is on python 3.9 without a good way to downgrade, so it would not work for Red. 
However, my limited testing suggested it's possible to work.

Overall, I'd say anyone asking about Termux should be told to wait until python 3.9 is supported before trying.

**Comment by Kowlin** [comment]
I'd honestly not recommend it by any means, it bring a lot of the problems on a system that is 100% not build with long running applications in mind, at the best its a neat little side project to see where one can stick Red on weird places. It'll work, but its no where near viable, where as one could simply buy a Pi 4(00) and get the same results in a way that are going to last. Also the fact that reputable Always-free providers give you already more raw horse power that an old phone can give one makes this essentially a no brainer for *anything* else.

**Reply by bobloy** [reply]
I agree with your comments, but I think you miss the most likely use case.

Being able to run on a phone is hypothetically a "free" hosting solution for those that can't or don't want to run on their personal computers 24/7.
Most people already have a phone, and rarely turn it off. 
I understand the "old phone" argument, but consider the young person on their first phone trying to host a bot with no money. 

Overall, I'd say this method of hosting would be highly sought after if there *were* a way to do it.

**Reply by aikaterna** [reply]
Back when I tried this previously a few years ago the bot would crash on a handoff between cell phone towers as there was technically no internet for a moment.
